torchvision/prototype/models/depth/stereo/crestereo.py

```diff
@@ -0,0 +1,1021 @@
+import math
+from functools import partial
+from typing import Iterable, List, Optional, Callable, Tuple, Dict, Union
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torchvision.models.optical_flow.raft as raft
+from torch import Tensor
+from torchvision.models._api import WeightsEnum
+from torchvision.models.optical_flow._utils import make_coords_grid, grid_sample, upsample_flow
+from torchvision.ops import Conv2dNormActivation
+
+
+class ResidualBlock(raft.ResidualBlock):
+    def __init__(self, in_channels, out_channels, *, norm_layer, stride=1):
+        super().__init__(in_channels, out_channels, norm_layer=norm_layer, stride=stride)
+
+        # the CREStereo base architecture changes the number of channels
+        # even on grids with the same spatial resolution
+        self.downsample = Conv2dNormActivation(
+            in_channels,
+            out_channels,
+            norm_layer=norm_layer,
+            kernel_size=1,
+            stride=stride,
+            bias=True,
+            activation_layer=None,
+        )
+
+
+class FeatureEncoder(raft.FeatureEncoder):
+    """Base encoder for Feature Encoder and Context Encoder"""
+
+    def __init__(
+        self,
+        *,
+        block: Callable[..., nn.Module] = ResidualBlock,
+        layers: Tuple[int, int, int, int, int] = (64, 64, 96, 128, 256),
+        strides: Tuple[int, int, int, int] = (2, 1, 2, 1),
+        norm_layer: Callable[..., nn.Module] = nn.InstanceNorm2d,
+    ):
+        super().__init__(block=block, layers=layers, strides=strides, norm_layer=norm_layer)
+        for s in strides:
+            if s not in [1, 2]:
+                raise ValueError(f"FeatureEncoder unsupported stride size {s}. Supported values are one of ``[1, 2]``.")
+
+        self.output_dim = layers[-1]
```

Comment:
That was actually used to dynamically construct to keys for the ModuleDicts. Although I do not expect someone to wish to change the default architecture, I just thought it would be nice to keep it in case someone might change the strides for the FeatureEncoder